### ISSUE TITLE
How to freeze the codebook? (maybe use `no_grad()` ?)

### ISSUE THREAD

**Issue by fyqqyf** [lucidrains/vector-quantize-pytorch#42]
I want to freeze the learned codebook, can I use the following code:
```
with torch.no_grad():
            quantized, _, _ = self.vq(x_group)
```
Looking forward to your reply.
Thanks!

**Comment by lucidrains** [comment]
you just have to set it to eval mode

**Comment by lucidrains** [comment]
@fyqqyf if you don't have any projections (codebook and input dimension are the same), then setting eval mode is enough (`vq_module.eval()`) otherwise your example code will be needed and will work fine
